vernac/declare.ml

```diff
@@ -1215,9 +1215,10 @@ let get_hide_obligations =
     ~key:["Hide"; "Obligations"]
     ~value:false
 
-let declare_obligation prg obl body ty uctx =
+let declare_obligation prg obl ~uctx ~types ~body =
+  let univs = UState.univ_entry ~poly:prg.prg_poly uctx in
   let body = prg.prg_reduce body in
-  let ty = Option.map prg.prg_reduce ty in
+  let ty = Option.map prg.prg_reduce types in
```

Comment:
TBH I prefer `typ` to `types` as there's only 1
Also why don't we shadow here?

Reply:
Shadowing done, I used `types` as this is what has been used on the rest of `Declare`; maybe #12372 would be a good opportunity to discuss a better naming overall.